src/api/utils/castutils.py

```diff
@@ -9,3 +9,20 @@ def to_decimal(val, max_digits, precision):
         ctx.prec = max_digits
         places = Decimal(10) ** (precision * -1)
         return Decimal(val).quantize(places, ROUND_HALF_DOWN)
+
+
+def to_number(string):
+    try:
+        num = float(string)
+        return int(num) if num.is_integer() else num
+    except ValueError:
+        return None
+
+
+def cast_str_value(string):
+    if not isinstance(string, str):
```

Comment:
Thinking we should combine and make consistent this check and those of `is_decimal` above